src/integer_mod_q/mat_zq/inverse.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright © 2023 Marcel Luca Schmidt
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains the implementation of the `inverse` function.
+
+use super::MatZq;
+use crate::{
+    integer::{MatZ, Z},
+    traits::{Concatenate, Gcd, GetNumRows},
+};
+use flint_sys::fmpz_mod_mat::fmpz_mod_mat_rref;
+
+impl MatZq {
+    /// Returns the inverse of the matrix if it exists (is square and
+    /// has a determinant unequal to zero) and `None` otherwise.
+    ///
+    /// Note that the modulus is assumed to be prime.
+    /// If it is not, it can happen that the function panics.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let mut matrix = MatZq::from_str("[[1,2],[3,4]] mod 7").unwrap();
+    /// let matrix_invert = matrix.inverse_prime().unwrap();
+    ///
+    /// let id = &matrix_invert * &matrix;
+    ///
+    /// assert_eq!("[[5, 1],[5, 3]] mod 7", matrix_invert.to_string());
+    /// assert_eq!("[[1, 0],[0, 1]] mod 7", id.to_string());
```

Comment:
```suggestion
    /// let mut matrix = MatZq::from_str("[[1,2],[3,4]] mod 7").unwrap();
    ///
    /// let matrix_invert = matrix.inverse_prime().unwrap();
    ///
    /// let id = &matrix_invert * &matrix;
    /// assert_eq!("[[5, 1],[5, 3]] mod 7", matrix_invert.to_string());
    /// assert_eq!("[[1, 0],[0, 1]] mod 7", id.to_string());
```
Make it more obvious where the function is called.